### PR TITLE
Localized License

### PR DESCRIPTION
**Pull request by sandesh556**
1. Install Billing from localizedlicense branch.
2. Place an .env file and set up the database. Change the App_url, DB_DATABASE_NAME, USERNAME, PASSWORD.
3. run php artisan migrate

### CODE REVIEW

**Review by bindu95 — APPROVED** [review]
https://github.com/ladybirdweb/faveo-helpdesk-advance/pull/10005
Updated the tested scenarios here.

File: app/Http/Controllers/Order/InvoiceController.php [hunk 1]
```diff
@@ -291,7 +291,7 @@ public function generateInvoice()
                 $this->createInvoiceItems($invoice->id, $cart);
             }
             if (emailSendingStatus()) {
-                $this->sendMail($user_id, $invoice->id);
+                // $this->sendMail($user_id, $invoice->id);
```
Comment (ashu555) [inline app/Http/Controllers/Order/InvoiceController.php:294]: Uncomment this

File: app/Http/Controllers/Order/RenewController.php [hunk 2]
```diff
@@ -69,9 +69,14 @@ public function renewBySubId($id, $planid, $payment_method, $cost, $code)
...
             $sub->save();
-            $licenseStatus = StatusSetting::pluck('license_status')->first();
-            if ($licenseStatus == 1) {
-                $this->editDateInAPL($sub, $updatesExpiry, $licenseExpiry, $supportExpiry);
+
+            if (Order::where('id', $sub->order_id)->value('license_mode') == 'File') {
+                Order::where('id', $sub->order_id)->update(['is_downloadable'=> 0]);
+            } else {
+                $licenseStatus = StatusSetting::pluck('license_status')->first();
+                if ($licenseStatus == 1) {
```
Comment (ashu555) [inline app/Http/Controllers/Order/RenewController.php:77]: No need to use comparison operator

**Review by ashu555 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Order/RenewController.php (hunk 2 above)
Comment (sandesh556) [reply app/Http/Controllers/Order/RenewController.php:77]: We need to specify the value it's not just boolean. it has 0,1 and 2 as values here

**Review by ashu555 — APPROVED** [review]

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Login as admin When he enters the domain in client order till public and then clicks on done it's redirecting to invoice page. 
image
last commit:-
image

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [x] Issue:- Not enter a domain name and click on done then download license file option gets disabled.
Verification scenarios remaining will try to finish tomorrow.

**Comment by sandesh556** [comment]
The done button bug for empty values has been removed long back. The client will be able to download again after the client submits an empty field. There is an error message shown if the client does that asking that person to try again.

**Review by SarfrazS — CHANGES_REQUESTED** [review]
suddenly billing not opening getting this error developer suggested do git clone and checkout to my branch and then test.
image

Note:- Sandesh shared with me scenarios tested working checking other scenarios in-between stop working billing.

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [x] Issue:- we are able to change license mode from faveoconfig and able to install how many installation limits is there in billing with the same license key.
image

- [x] Query:- Why we are downloading license files and private keys and uploading under the signature folder. How customer will know where I should upload.
- [x] Query:- Client able to download the product from the order download option or not in live billing, if yes then what is the use case for localized, if not then how a client will download product.(here I m not able to download because path issue)

**Comment by sandesh556** [comment]
Issue resolved i have integrated the installations of license manager and localized.

Query 1: There's a tooltip right next to the download buttons which suggest where to place the downloaded files

Query 2: Yes Client will be able to download products using live billing since the product files are not present in test billing. 

Use case of the localized license is if the client has an issue hitting Faveo license manager server(can be due to a firewall or just not willing to hit) we enable a localized license which is a safely encrypted license file that can only be decrypted using the private key which is different for every user and hence the client can use Faveo with most of the checks that license manager performs using localized so our product is not given without a license.

**Review by SarfrazS — APPROVED** [review]
Functionality working fine. 
Sandesh shared with me to test these scenarios all scenarios checked working fine.
After that, I checked other scenarios also working fine.

**Review by mverma16 — APPROVED** [review]

**Review by mverma16 — APPROVED** [review]

**Pull request merged**